robotoff/spellcheck/patterns/__init__.py

```diff
@@ -0,0 +1,38 @@
+from typing import Dict
+from pathlib import Path
+
+from robotoff.spellcheck.base_spellchecker import BaseSpellchecker
+
+Patterns = Dict[str, str]
+
+
+class PatternsSpellchecker(BaseSpellchecker):
+
+    PATTERNS_PATHS = {"fr": Path(__file__).parent / "patterns_fr.txt"}
+
+    def __init__(self, lang: str = "fr"):
+        self.lang = lang
+        self.patterns: Patterns = {}
+
+        with self.PATTERNS_PATHS[lang].open() as f:
+            current_pattern = None
+            for line in f.readlines():
```

Comment:
`for line if f` works as well, and it iterates over line in the file, contrary to readlines() which load all file content in memory

Reply:
Interesting, thanks !
Changed in https://github.com/openfoodfacts/robotoff/pull/181/commits/b37abc29a8268aa8b04d3db87e5780b7ca138709